Experiment automation/(LUMICKS) Reference kit automation script/Catch_beads_DNA_record_Kymo_for_LacI_version_1.py

```diff
@@ -0,0 +1,318 @@
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+--------------------------------
+This script Iteratively catches beads, fishes for DNA and makes a kymograph at 10 pN.
+The script is optimized for DNA tethers with a length of 37.8 kbp (12.85 micron).
+Before running the script
+    1) The traps have to be aligned horizontally.
+    2) Calibrate the force
+    3) Set the scan options to a continuous kymograph and configure the kymograph,
+    ie make sure the correct location is being scanned when making a kymograph
+    4) Make sure the traps are not too close together for catching beads
+
+Workflow of the script
+    1) Catch beads
+    2) Catch a DNA tether
+    3) Check if it is only a single tether, if not, try to break tethers until there is 1 left.
+    4) Move using the microstage to 'J1' at 60% speed
+    5) Move using the microstage to protein channel at 60% speed
+    6) Set the force to 10 pN
+    7) Start a kymograph.
+    8) Stop the kymograph.
+    9) Restart step 1 to 8 n times, where n is the number of kymographs you want to make.
+"""
+from datetime import datetime
+import bluelake as bl
+import time
+import os
+import numpy as np
+
+
+name_bead_channel = "beads"  # This has to match the correct channel name in the UI
+name_dna_channel = "DNA"  # This has to match the correct channel name in the UI
+name_protein_channel = "Ch4"
+name_junction = "J1"
+name_buffer_channel = "buffer"
+time_to_wait_for_flow = 5.0
+dna_length = 0.34*37.8  # Length of DNA in micrometers
+
+distance = bl.timeline["Distance"]["Distance 1"]
+force = bl.timeline["Force LF"]["Trap 2"]
+match_score1 = bl.timeline["Tracking Match Score"]["Bead 1"]
+match_score2 = bl.timeline["Tracking Match Score"]["Bead 2"]
+
+
+def throw_if_beads_lost(match_threshold):
+    """Raise an exception if we lose the beads."""
+    if match_score1.latest_value < match_threshold or match_score2.latest_value < match_threshold:
+        raise RuntimeError("Lost beads")
+
+
+def set_pressure(target):
+    """Increase pressure until we are above a certain target level"""
+    while bl.fluidics.pressure < target:
+        bl.fluidics.increase_pressure()
+        bl.pause(1.0)
+
+
+def start_flow(pressure):
+    print("Starting Flow.")
+    set_pressure(pressure)
+    bl.fluidics.open(1, 2, 3, 6)
+
+
+def stop_flow():
+    print("Stopping flow.")
+    bl.fluidics.close(1, 2, 3, 6)

```

Comment:
Do you ever expect these channel numbers to change? Might be worthwhile to define them in one place, so someone doesn't change it in only one place and forgets the other by accident.